migtests/scripts/functions.sh

```diff
@@ -501,59 +505,84 @@ import_data_file() {
     yb-voyager import data file ${args} $*
 }
 
-
 archive_changes() {
-	ENABLE=$(shuf -i 0-1 -n 1)
-	echo "archive changes ENABLE=${ENABLE}"
-	if [[ ${ENABLE} -eq 1 ]];
-	then
-		ARCHIVE_DIR=${EXPORT_DIR}/archive-dir
-		mkdir ${ARCHIVE_DIR}  # temporary place to store the archive files
-
-		yb-voyager archive changes --move-to ${ARCHIVE_DIR} \
-		--export-dir ${EXPORT_DIR} \
-		--fs-utilization-threshold 0
-	fi
+    ENABLE=$(shuf -i 0-1 -n 1)
+    echo "archive changes ENABLE=${ENABLE}"
+    if [[ ${ENABLE} -eq 1 ]]; then
+        if [ "${run_via_config_file}" = "true" ]; then
+            # Run using the generated config file
+            yb-voyager archive changes -c "${GENERATED_CONFIG}" --yes
+        else
```

Comment:
nit nit: avoid nested if else, use early return

https://medium.com/softwarecraft-mastery/the-return-early-pattern-c554f47fe58a